eth/filters/filter.go

```diff
@@ -142,6 +148,11 @@ func (f *Filter) Logs(ctx context.Context) ([]*types.Log, error) {
 	if f.end == -1 {
 		end = head
 	}
+
+	if f.rangeLimit && (f.end-f.begin) > maxFilterBlockRange {
```

Comment:
Instead of rejecting the request directly, can we somehow cap the filter range but still notify users about the capping?

Reply:
Modified. Could you clarify what you have in mind about notifying the users?